### DISCUSSION TITLE
Offline/occasionally connected functionality

### DISCUSSION THREAD

**Discussion by NCC-DAVID-HARRIS** [Budibase/budibase#2625] [Ideas]
Hi there,

I am just looking round for form building solutions and this project looks great!

One requirement we have is the ability to have offline form/data gathering.
Is is due to we are not guaranteed that the device being used will have connectivity 100% of the time.

I have taken a look round and haven't found any documentation if Budibase can do this.

Can Budibase do offline/occasionally connected data gathering?
Any links to how-tos or documentation appreciated!

Thanks!

**Comment by mjashanks** [comment]
Hi There.

We cannot currently do this in Budibase. However, we do get asked for this **and** budibase's architecture actually supports this really well. We are based on CouchDB + PouchDB which have great offline sync capabilities.

Watch this space! But, won't be here in the next few months for sure.

**Reply by pedro-psb** [reply]
I have a very similiar use case. Are there any news on this?

**Reply by benaunt** [reply]
This will be of significant benefit to our needs as long as it works on the self hosted version.

**Reply by dwoldo** [reply]
Agreed, we often deploy into disconnected, intranet only environments and a self-hosted version would be ideal.

**Reply by JCV67** [reply]
Hello,
Is there any development on this?

**Reply by StudioRav** [reply]
Does this work now?

**Reply by ConorWebb96** [reply]
Hey all,

I'm afraid we're still unable to accommodate this request at the moment. Additionally, it's not currently on our roadmap, so it's unlikely to receive priority in the near future.
